contrib/scalar/scalar.c

```diff
@@ -278,37 +322,57 @@ static int cmd_clone(int argc, const char **argv)
 
 	/* TODO: handle local cache root */
 
-	/* TODO: check whether to use the GVFS protocol */
+	if (!branch &&
+	    !(branch = remote_default_branch(dir, url))) {
+		res = error(_("failed to get default branch for '%s'"), url);
+		goto cleanup;
+	}
 
 	config_path = xstrfmt("%s/.git/config", dir);
 
-	/* TODO: this should be removed, right? */
-	/* protocol.version=2 is broken right now. */
-	if (set_config(config_path, "protocol.version=1") ||
+	strbuf_reset(&buf);
+	strbuf_addf(&buf, "remote.origin.fetch="
+		    "+refs/heads/%s:refs/remotes/origin/%s",
+		    single_branch ? branch : "*",
+		    single_branch ? branch : "*");
+	if (set_config(config_path, "core.useGVFSHelper=false") ||
 	    set_config(config_path, "remote.origin.url=%s", url) ||
 	    /* TODO: should we respect single_branch here? */
-	    set_config(config_path, "remote.origin.fetch="
-		       "+refs/heads/*:refs/remotes/origin/*") ||
+	    set_config(config_path, buf.buf) ||
 	    set_config(config_path, "remote.origin.promisor=true") ||
 	    set_config(config_path,
-		       "remote.origin.partialCloneFilter=blob:none"))
-		return error(_("could not configure '%s'"), dir);
+		       "remote.origin.partialCloneFilter=blob:none")) {
+		res = error(_("could not configure '%s'"), dir);
+		goto cleanup;
+	}
 
 	if (!full_clone &&
-	    (res = run_git(dir, "-c", "core.useGVFSHelper=false",
-			   "sparse-checkout", "init", "--cone", NULL)))
+	    (res = run_git(dir, "sparse-checkout", "init", "--cone", NULL)))
 		goto cleanup;
 
 	if (set_recommended_config(config_path))
 		return error(_("could not configure '%s'"), dir);
 
+	if (cache_server_url ||
+	    !retrieve_cache_server_url(dir, url, &cache_server_url)) {
+		if (set_config(config_path, "core.useGVFSHelper=true") ||
+		    set_config(config_path, "remote.origin.promisor") ||
+		    set_config(config_path,
+			       "remote.origin.partialCloneFilter") ||
```

Comment:
I don't think these apply at the same time as `core.gvfs`.

Reply:
Also, we should still use `core.useGVFSHelper=true` and `core.gvfs=150` when the `/gvfs/config` endpoint exists. We can use the origin instead of the cache server. This would show up with the Scalar functional tests because the test repo doesn't have a cache server.

This presents an interesting step of this effort: make the scalar functional tests consume this feature branch!

Reply:
I probably should clarify that this syntax (where there is no `=` in the value) _unsets_ the variable.

But I agree, it should not even be set in the first place. Will re-do it.

Reply:
I tried to address this via cca26c13cef. What do you think?